### ISSUE TITLE
Allow agnostic dependency discovery + link depending on enabled importers/exporters

### ISSUE THREAD

**Issue by SpaceIm** [assimp/assimp#3213]
I'm trying to improve conan recipe of Assimp (in order to have a better dependency graph).

Currently, without Hunter, Assimp relies on embedded codes for most of its dependencies (I don't take into account `gtest` or `stb` which are only used for tests or samples):

External dependencies (fallback to embedded one if not found):
- zlib
- IrrXML

Embedded dependencies:
- minizip
- utfcpp
- zip
- rapidjson
- poly2tri
- openddlparser
- polyclipping (old version 4.8.8, not API compatible with latest major version)
- Open3DGC

Except for `Open3DGC`, Assimp CMakeLists could allow -through options- consumers to find these libraries in order to build Assimp, and eventually fallback to embedded one.
I don't know much about Hunter, but all of this should be "package manager agnostic" (use of `find_package`).

Moreover, some of these dependencies are not always needed, but Assimp compile them (or link them if Hunter) unconditionally. Seems like:
- `zip` is only required for `3MF` Exporter
- `rapidjson` is only required for `glTF` and `glTF2`
- `poly2tri` is only required for `Blend` and `IFC` Importers
- `polyclipping` is only required `IFC` Importers
- `openddlparser` is only required for `OpenGEX` Importer
- `Open3DGC` is only required for `glTF`
- `zlib` is only required for `ASSBIN` importer and exporter, `ASSXML` exporter, `Blend`, `FBX`, `q3bsp`, `X` and `XGL` importers

Finally, I think that `zip` library is dangerous because it implements a modified version of `miniz` header-only version, which might lead to duplicated symbols if consumer depends on `Assimp` and `miniz`.

**Comment by kimkulling** [comment]
You are right, there is a cleanup needed to have only one zip-file. A plan could be to use only an abstraction for the different apis.

**Comment by jdumas** [comment]
+1 to this. I just had a conflict between Assimp's embedded `zip` library and the `miniz` embedded in tinyexr. Currently we can't even disable `zip`, even if we disable the 3MF importer...

**Comment by tellypresence** [comment]
@jdumas Is this still a problem?

**Comment by jdumas** [comment]
Unless something has changed recently, I'd say yes definitely.

**Comment by tellypresence** [comment]
Build issues tracked via epic #6189
